src/account/operations/participation/mod.rs

```diff
@@ -42,52 +42,58 @@ impl AccountHandle {
 
-        for output_data in participation_outputs {
-            // PANIC: the filter already checks that the metadata exists.
-            let metadata = output_data.output.features().and_then(|f| f.metadata()).unwrap();
-            // TODO don't really need to collect if we only need the first
-            let event_ids = if let Ok(participations) = Participations::from_bytes(&mut metadata.data()) {
-                participations
-                    .participations
-                    .into_iter()
-                    .map(|p| p.event_id)
-                    .collect::<Vec<ParticipationEventId>>()
-            } else {
-                // No valid participation in this output, skip it.
-                continue;
-            };
+        for output_data_chunk in participation_outputs.chunks(100).map(|x| x.to_vec()) {
+            let mut tasks = Vec::new();
+            for output_data in output_data_chunk {
+                // PANIC: the filter already checks that the metadata exists.
+                let metadata = output_data.output.features().and_then(|f| f.metadata()).unwrap();
+                // TODO don't really need to collect if we only need the first
+                let event_ids = if let Ok(participations) = Participations::from_bytes(&mut metadata.data()) {
+                    participations
+                        .participations
+                        .into_iter()
+                        .map(|p| p.event_id)
+                        .collect::<Vec<ParticipationEventId>>()
+                } else {
+                    // No valid participation in this output, skip it.
+                    continue;
+                };
 
-            // TODO: which client to use if there are multiple events and one node isn't tracking all of them?
-            let event_client = if let Some(event_id) = event_ids.first() {
-                self.get_client_for_event(event_id).await?
-            } else {
-                self.client().clone()
-            };
-
-            if let Ok(status) = event_client.output_status(&output_data.output_id).await {
-                for (event_id, participation) in status.participations {
-                    match participations.entry(event_id) {
-                        Entry::Vacant(entry) => {
-                            entry.insert(HashMap::from([(output_data.output_id, participation)]));
-                        }
-                        Entry::Occupied(mut entry) => {
-                            entry.get_mut().insert(output_data.output_id, participation);
+                // TODO: which client to use if there are multiple events and one node isn't tracking all of them?
+                let event_client = if let Some(event_id) = event_ids.first() {
+                    self.get_client_for_event(event_id).await?
+                } else {
+                    self.client().clone()
+                };
+
+                tasks.push(async move {
```

Comment:
I wonder if we really need this extraneous block, it just seems to create a future for nothing?

Reply:
For some reason it needs to be required to be really executed in parallel
```Rust
 for (index, output_data) in output_data_chunk.into_iter().enumerate() {
 ...
                tasks.push(
                    tokio::spawn(async move { 
                        println!("Index: {index}");
                        (event_client.output_status(&output_data.output_id).await, output_data) })
                        
                );
```
will run them all in order 
```
Index: 0
Index: 1
Index: 2
Index: 3
Index: 4
Index: 5
Index: 6
Index: 7
Index: 8
Index: 9
Index: 10
Index: 11
...
```
With the extra future it will be
```
index: 2
index: 7
index: 5
index: 20
index: 13
index: 4
index: 14
index: 8
index: 3
index: 0
index: 15
index: 6
index: 1
index: 10
```

Reply:
Interesting, I'm not sure to understand what's happening here but ok